### DISCUSSION TITLE
Display discord name on card instead discord Id

### DISCUSSION THREAD

**Discussion by showdance** [StudyLions/StudyLion#36] [Ideas]
Hello all,
I wanted to ask if it's possible for Lion bot to display user 'discord name' instead 'discord id' on the profile card or have option to customize it?
So for example for user to see:
DISCORD_DISPLAY_NAME#xxxx instead DISCORD_USER_ID#xxxxx.

I can help with making this change and create Pull Request for it, I' backend software engineer.

**Comment by Intery** [comment]
Hi showdance,

Thanks for your offer of contribution! New contributors are always welcome.

Regarding your issue, showing the `userid` on the profile card (or elsewhere) is actually a last-ditch fallback for when Leo cannot find the user in cache, and has not saved a username for them. This should be an extremely rare issue, caused by the internal member cache going out of sync with the server member list, usually due to Discord not sending the member information in the first place. After Leo has cached the member and run a command from them for the first time, their username should hopefully be saved and displayed in all relevant UI elements.
